build.gradle

```diff
@@ -52,19 +60,16 @@ task buildWebClient(type: Exec) {
     commandLine 'npm', 'run', 'build'
 }
 
-jib.extraDirectory = file('build/extra-directory')
-
 task copyWebClient(type: Copy) {
     from file('web-client/dist/container-jfr-web')
-    into file('build/extra-directory/web-client')
+    into file("${buildDir}/generated/sources/web-client/resources/main/com/redhat/rhjmc/containerjfr/net")
```

Comment:
Hmm. I get that this puts the assets somewhere where the bootstrap classloader will find them, but it seems strange to put non-sources like PNGs, TTFs, and the sort-of-source-built-artefacts like actual HTML/CSS/JS of the client into "generated sources". Is there anywhere else that the assets can be placed which will end up with them searchable on the classpath? Or, can a new location be specified for them and added to the search classpath?

Reply:
In theory, is location can be anywhere, as long as the paths in `copyWebClient` and `sourceSets` are consistent. I put them under `generated` as I thought they are, indeed, resources generated by another program (`npm`) from external sources (`container-jfr-web`).

Anyway, the location can be arbitrary, but they will be eventually copied to `build/resources`, by `processResources` task, for the bootstrap class loader to discover. Any idea where you want it to be?

Reply:
I suppose. `generated/sources` sounds to me more like Java sources, ex. stuff produced by Dagger for direct injection and the likes. These things themselves aren't source files though, especially not after they've been processed through the npm build.

Something like `${buildDir}/assets` might do?